api/include/opentelemetry/trace/noop.h

```diff
@@ -55,11 +60,11 @@ class NoopSpan final : public Span
 
   bool IsRecording() const noexcept override { return false; }
 
-  SpanContext GetContext() const noexcept override { return span_context_; }
+  SpanContext GetContext() const noexcept override { return *span_context_.get(); }
```

Comment:
```suggestion
  SpanContext GetContext() const noexcept override { return span_context_->get(); }
```

Reply:
This is what we are doing here : `*(span_context.get())` . Don't think `span_context_->get()` will work.

Reply:
@ThomsonTan - I hope you are fine with the comment. Else, we will resolve this in another PR.